lib/diagnostic.js

```diff
@@ -5,17 +5,31 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current) {
+  let result = 0;
+  for (let i = 0; i < previous.length; i++) {
```

Comment:
```previous``` and ```current``` are assumed to be numbers and since ```number``` doesn't have a ```length``` property, ```previous.length``` will return ```undefined``` and the loop will never run.